docs/tla/ThreePhase.tla

```diff
@@ -0,0 +1,77 @@
+(* MIT License Copyright (c) 2020 CQFN
+ *    https://github.com/cqfn/degitx/blob/master/LICENSE
+ *)
+------------------------------ MODULE ThreePhase ------------------------------
+CONSTANT RM
+
+VARIABLES rmState, tmState, tmPrepared, msgs
+
+Messages == [type: {"Prepared"}, rm: RM] \union [type:{"PreCommit", "Commit", "Abort"}]
+
+\* All possible states of Resource Managers and Transaction Managers
+TPTypeOK == /\ rmState \in [RM -> {"working", "prepared", "pre-committed", "committed", "aborted"}]
+            /\ tmState \in {"init", "done"}
```

Comment:
TmState of 3PC is a bit complexer, please see Figure 7 of "Nonblocking commit protocols , by Dale Skeen". There are "init", "working", "prepared", "committed", and "aborted" states of TM